integreat_cms/cms/utils/text_utils.py

```diff
@@ -17,3 +17,24 @@ def lowfirst(string):
     :rtype: str
     """
     return string and str(string)[0].lower() + str(string)[1:]
+
+
+def truncate_bytewise(string, length):
+    """
+    Truncate a string to a maximum byte length.
+
+    :param string: The input text
+    :type string: str
+
+    :param length: The maximum length of the text in byte representation
+    :type length: int
+
+    :return: The truncated text
+    :rtype: str
+    """
+    encoded_string = string.encode()
+    try:
+        print(len(encoded_string[:length]))
+        return encoded_string[:length].decode()
+    except UnicodeDecodeError as e:
```

Comment:
IMHO we could actually implement a correct parsing of the UTF-8 bytes, which is not that difficult: https://en.wikipedia.org/wiki/UTF-8#Encoding

We just need to jump to the position indicated by the `length` parameter and bytewise go back until the byte does not start with `10xxxxxx`. The check could be implemented like this: `if myByte&1 == 1 and myByte&2 == 0:`. I'm not a bit wizard though and there may be even better solutions ;)